components/remote_settings/src/client.rs

```diff
@@ -122,24 +122,29 @@ pub trait ApiClient {
 
 /// Client for Remote settings API requests
 pub struct ViaductApiClient {
+    /// Server base URL
+    ///
+    /// This is something like `https://[domain]/v1/`.  It's what's normally called the `base_url`
+    /// in other client implementations.
+    server_url: Url,
     /// Base URL for requests to a collections endpoint
     ///
     /// This is something like
-    /// `https://[server-url]/v1/buckets/[bucket-name]/collections/[collection-name]/"
+    /// `https://[server-url]/buckets/[bucket-name]/collections/[collection-name]/`
     ///
-    /// Note: this is different than the `base_url` used for other client implementations (
-    /// (`https://[server-url]/v1).  The main reason to use the collection_url is that we can use
-    /// it to check if we need to invalidate the cached data stored in the [Storage] layer.
+    /// The reason to store collection_url is that we can use it to check if we need to invalidate
+    /// the cached data stored in the [Storage] layer.
     collection_url: Url,
     remote_state: RemoteState,
 }
 
 impl ViaductApiClient {
     fn new(server_url: Url, bucket_name: &str, collection_name: &str) -> Result<Self> {
         let collection_url = server_url.join(&format!(
-            "v1/buckets/{bucket_name}/collections/{collection_name}/"
+            "buckets/{bucket_name}/collections/{collection_name}/"
```

Comment:
```suggestion
            "/buckets/{bucket_name}/collections/{collection_name}/"
```

Reply:
I avoided this because it's going to break the join.  Without the trailing slash, then join treats the last component as a file rather than a directory and removes it.  However, there is the `path_segments_mut()` method.  Let me try that and see how it comes out.